prov/shm/src/smr_ep.c

```diff
@@ -1387,6 +1393,13 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 		if (smr_env.use_dsa_sar)
 			smr_dsa_context_init(ep);
 
+		/* if XPMEM is on after exchanging peer info, then set the
+		 * endpoint p2p to XPMEM so it can be used on the fast
+		 * path
+		 */
```

Comment:
does exchanging peer info have anything to do with this? I think I would prefer the p2p_type to be set in a single line with the default case

Reply:
The idea here is that it's potentially possible to turn off xpmem if any of the peers do not support xpmem. So this would be the most accurate place to set XPMEM type. The idea is all peers should support xpmem or none of them will use it.

Reply:
can you please tell me what you want to see. I'm not following your comment.

Reply:
we set the default to CMA and set it to XPMEM if XPMEM is on. When we exchange the peer information at that point we try to enable xpmem for that peer and if we fail we turn off xpmem locally as well. The intent here is if we can't talk to any of the peers over xpmem, then no one talks over xpmem and we fall back to cma. This behavior makes sense since XPMEM is system wide and if it's enabled on the node all peers should be able to use it and if there is a failure then no one uses it.

This is the reason I put it here.